frame/contracts/src/exec.rs

```diff
@@ -941,6 +984,28 @@ where
 		result
 	}
 
+	fn delegate_call(
+		&mut self,
+		code_hash: CodeHash<Self::T>,
+		input_data: Vec<u8>,
+	) -> Result<ExecReturnValue, ExecError> {
+		let executable = E::from_storage(code_hash, &self.schedule, self.gas_meter())?;
+		let top_frame = self.top_frame_mut();
+		let contract_info = top_frame.contract_info().clone();
+		let account_id = top_frame.account_id.clone();
+		let value = top_frame.value_transferred.clone();
+		let executable = self.push_frame(
+			FrameArgs::Call {
+				dest: account_id,
+				cached_info: Some(contract_info),
+				delegated_call: Some(DelegatedCall { executable, caller: self.caller().clone() }),
+			},
+			value,
+			0,
```

Comment:
Why do we set the `gas_limit` to `0` here?

Reply:
We want to pass all the gas to the callee contract.

Reply:
That one is obvious. But why?

Reply:
The same way it is done in [ethereum](https://github.com/openethereum/parity-ethereum/blob/55c90d4016505317034e3e98f699af07f5404b63/ethcore/evm/src/interpreter/mod.rs#L607).
The idea is that it should be treated as the same call, with the same amount of gas, same caller, same value.

Reply:
Hmm but the EVM specifies that you can pass a gas parameter: https://www.ethervm.io/#F4

Reply:
So from what I could decypher from the [Yellow Paper p.38](https://ethereum.github.io/yellowpaper/paper.pdf) it looks like `DELEGATECALL` just takes the remaining gas at that point in the execution (so it "keeps" the context in that sense)

Reply:
Ahh okay thanks for looking this up.

Reply:
@yarikbratashchuk do you mind adding a comment about this in the code for future readers?